src/components/Tiles/MovieTile.js

```diff
@@ -1,12 +1,17 @@
 import { Star } from '../Star/Star';
 import { ImageComponent } from '../ImageComponent/ImageComponent';
 import ApiProvider from '../../providers/ApiProvider';
+import { observerListener } from '../../shared/observer';
 
 export const MovieTile = async ({ movieID }) => {
     const movie = await ApiProvider.getTitleDetails(movieID);
 
+    observerListener(`movie-tile-${movie.id}`, () => {
+        window.location.href = `/details?id=${movie.id}`;
```

Comment:
1. We have redirect function, it should be used instead
2. We have object routes which should be used unstead of hardcoded route
3. There is an ImageComponent which is clickable and is receiving the onClick function as an argument, can't we use that component here instead of this whole logic for adding listener?

Reply:
1. done
2. done
3. i didnt find way to use this solution.